server/middleware/users.ts

```diff
@@ -0,0 +1,21 @@
+import { ForbiddenError } from 'express-response-errors';
+import DOMpurify from 'isomorphic-dompurify';
+
+import { AnyRequestHandler } from 'server/types/express';
+
+export const requireSameId: AnyRequestHandler = async (req, res, next) => {
+  if (req.user.dataValues.id.toString() !== req.params.id) {
+    throw new ForbiddenError('User can only make changes to their own profile');
+  }
```

Comment:
just curious, if you had done `req.user.id.toString()` would that have worked as well?